ext/flash/01_http.js

```diff
@@ -353,183 +540,37 @@
-                      "op_flash_write_resource",
-                      http1Response(
-                        method,
-                        innerResp.status ?? 200,
-                        innerResp.headerList,
-                        0, // Content-Length will be set by the op.
-                        null,
-                        true,
-                      ),
-                      serverId,
-                      i,
-                      resourceRid,
-                    ).then(() => {
-                      // Release JS lock.
-                      readableStreamClose(respBody);
-                    });
-                  } catch (error) {
-                    await reader.cancel(error);
-                    throw error;
-                  }
-                } else {
-                  const reader = respBody.getReader();
-                  writeFixedResponse(
+              resp = handler(req);
+              if (resp instanceof Promise || resp.then) {
+                resp.then((resp) =>
+                  handleResponse(
+                    req,
+                    resp,
+                    body,
+                    hasBody,
+                    method,
                     serverId,
                     i,
-                    http1Response(
-                      method,
-                      innerResp.status ?? 200,
-                      innerResp.headerList,
-                      respBody.byteLength,
-                      null,
-                    ),
-                    respBody.byteLength,
-                    false,
                     respondFast,
-                  );
-                  while (true) {
-                    const { value, done } = await reader.read();
-                    await respondChunked(
-                      i,
-                      value,
-                      done,
-                    );
-                    if (done) break;
-                  }
-                }
-              }
-
-              if (ws) {
-                const wsRid = await core.opAsync(
-                  "op_flash_upgrade_websocket",
-                  serverId,
-                  i,
-                );
-                ws[_rid] = wsRid;
-                ws[_protocol] = resp.headers.get("sec-websocket-protocol");
-
-                ws[_readyState] = WebSocket.OPEN;
-                const event = new Event("open");
-                ws.dispatchEvent(event);
-
-                ws[_eventLoop]();
-                if (ws[_idleTimeoutDuration]) {
-                  ws.addEventListener(
-                    "close",
-                    () => clearTimeout(ws[_idleTimeoutTimeout]),
-                  );
-                }
-                ws[_serverHandleIdleTimeout]();
+                    respondChunked,
+                  )
+                ).catch(onError);
+                continue;
               }
-            })().catch(onError);
+            } catch (e) {
+              resp = await onError(e);
```

Comment:
Found an edgecase: Because this line is not going through the same treatment a users custom onError method can block future requests.

This will only happen if the handler threw an error outside of a promise (example below).


```
const handler = (request) => {
  if (Math.random() > .5) throw new Error('blow up randomly');
  return (async() => {
    return new Response('handler can still return a promise');
  })();
}
```


Alternatively, something like this could be done. let me know what you think.
```typescript
Promise.resolve()
  .then(() => handle(req))
  .catch(error => onError(error))
  .then(resp => handleResponse(...));

```